src/AspNet.Security.OAuth.Weixin/WeixinAuthenticationHandler.cs

```diff
@@ -115,35 +126,61 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OA
 
     protected override string BuildChallengeUrl([NotNull] AuthenticationProperties properties, [NotNull] string redirectUri)
     {
-        string stateValue = Options.StateDataFormat.Protect(properties);
+        var scopeParameter = properties.GetParameter<ICollection<string>>(OAuthChallengeProperties.ScopeKey);
+        var scope = scopeParameter != null ? FormatScope(scopeParameter) : FormatScope();
+
+        var parameters = new Dictionary<string, string?>()
+        {
+            ["appid"] = Options.ClientId,
+            ["scope"] = scope,
+            ["response_type"] = "code",
```

Comment:
Nit: unnecessary `,`.

Reply:
Again personal choice - I prefer to _always_ have the comma because:

1. Consistency
2. Adding a new item is a 1 line change, not a 2 line change

If you feel strongly about it I can take them back out.

Reply:
I find it a bit weird but I'm not strongly opposed to adding a `,` on each line if you prefer 😃
That said, things will now be inconsistent if you don't adopt this code style everywhere, which is not ideal.

Reply:
I've made the others match to have a trailing `,`, and which got you some more bonus `var` usage 😄